jcl/src/java.base/share/classes/jdk/crac/Context.java

```diff
@@ -0,0 +1,89 @@
+/*[INCLUDE-IF CRAC_SUPPORT]*/
+/*******************************************************************************
+ * Copyright IBM Corp. and others 2024
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ *******************************************************************************/
+package jdk.crac;
+
+import java.nio.file.Paths;
+
+import openj9.internal.criu.InternalCRIUSupport;
+import openj9.internal.criu.JVMCheckpointException;
+import openj9.internal.criu.JVMRestoreException;
+
+/**
+ * A Resource that allows other Resources to be registered with it.
+ */
+public abstract class Context<R extends Resource> {
+
+	final static boolean debug = System.getProperty("enable.jdk.crac.api.debug") != null; //$NON-NLS-1$
+
+	static Context<Resource> globalContext = new CRIUSupportContext<>();
+
+	/**
+	 * @param resource a Resource to be registered
+	 */
+	public abstract void register(R resource) throws Exception;
+
+	static Context<Resource> createGlobablContext() {
+		return globalContext;
+	}
+
+	public void checkpointJVM() {
```

Comment:
I dont think Context has a public `checkpointJVM` function

Reply:
it looks like

```
public abstract class jdk.crac.Context<R extends jdk.crac.Resource> implements jdk.crac.Resource {
  protected jdk.crac.Context();
  public abstract void beforeCheckpoint(jdk.crac.Context<? extends jdk.crac.Resource>) throws jdk.crac.CheckpointException;
  public abstract void afterRestore(jdk.crac.Context<? extends jdk.crac.Resource>) throws jdk.crac.RestoreException;
  public abstract void register(R);
}
```

Reply:
Matched `Context` methods.